networkx/algorithms/operators/unary.py

```diff
@@ -67,3 +68,46 @@ def reverse(G, copy=True):
         raise nx.NetworkXError("Cannot reverse an undirected graph.")
     else:
         return G.reverse(copy=copy)
+
+
+@not_implemented_for('directed')
+@not_implemented_for('multigraph')
+def mycielski(G, iterations=1):
+    """Returns the Mycielskian of a simple, undirected graph G
+
+    The Mycielskian of graph preserves a graph's triangle free
+    property while increasing the chromatic number by 1.
+
+    Parameters
+    ----------
+    G : graph
+        A simple, undirected NetworkX graph
+    iterations : int
+        The number of iterations of the Mycielski operation to
+        preform on G. Defaults to 1. Must be a non-negative integer.
+
+    Returns
+    -------
+    M : graph
+        The Mycielskian of G after the specified number of iterations.
+
+    Notes
+    ------
+    Graph, node, and edge data are not necessarily propagated to the new graph.
+
+    """
+    if not isinstance(iterations, int) or iterations < 1:
```

Comment:
No need to check that the number of iterations is an integer, that's too much type checking.